src/txn/txn.c

```diff
@@ -881,23 +881,33 @@ __txn_commit_timestamps_usage_check(WT_SESSION_IMPL *session, WT_TXN_OP *op, WT_
           "no timestamp provided for an update to a "
           "table configured to always use timestamps once they are first used"));
 
-    if (FLD_ISSET(ts_flags, WT_DHANDLE_TS_ORDERED) && txn_has_ts && prev_op_durable_ts > op_ts)
+    /*
+     * The program should exit abnormally in addition to just logging a message.
+     */
+    if (FLD_ISSET(ts_flags, WT_DHANDLE_TS_ORDERED) && txn_has_ts && prev_op_durable_ts > op_ts) {
         WT_RET(__wt_msg(session,
           WT_COMMIT_TS_VERB_PREFIX
           "committing a transaction that updates a "
           "value with an older timestamp (%s) than is associated with the previous "
           "update (%s) on a table configured for strict ordering",
           __wt_timestamp_to_string(op_ts, ts_string[0]),
           __wt_timestamp_to_string(prev_op_durable_ts, ts_string[1])));
+        WT_ASSERT(session, op_ts > prev_op_durable_ts);
```

Comment:
We can assert unconditionally if we entered this if block.
```
WT_ASSERT(session, false);
```

Reply:
Thanks for pointing that out, I'll do this instead.